dashi/poolqc/pooling_qc_sample.py

```diff
@@ -216,35 +291,87 @@ def update_sampleindices(run):
     return {
         'data': data,
         'layout': {
-            'title': 'Index Clusters per Sample. Passed Samples: %s Threshold: %s' % (
-                samples_passing_clusters, INDEX_THRESHOLD),
-            'xaxis': {'title': 'Sample', 'automargin': True},
+            'title': 'Index Clusters per Sample',
+            'xaxis': {'title': 'Sample', 'automargin': True, 'tickangle': '90', },
             'yaxis': {'title': 'Clusters'},
             'showlegend': False,
+            'barmode': 'group',
 
         }
     }
 
 
+def percent_difference(run, index_threshold):
+    run = run.sort_values('library')
+
+    data = []
+
+    for inx, d in run.groupby('library'):
+        d['Per Cent Difference'] = (d['Clusters'] - index_threshold) / index_threshold * 100
+        data.append(
+            go.Bar(
+                x=list(d['library']),
+                y=list(d['Per Cent Difference']),
+                name=inx,
+                marker={'color': '#20639B'},
+            )
+        )
+    return {
+        'data': data,
+        'layout': {
+            'title': 'Per Cent Difference of Index Clusters',
+            'xaxis': {'title': 'Sample', 'automargin': True, 'tickangle': '90', },
+            'yaxis': {'title': 'Per Cent', 'range': [-100, 100]},
+            'showlegend': False,
+        }
+    }
+
+
 @app.callback(
     [dep.Output('SampleIndices', 'figure'),
+     dep.Output('Per Cent Difference', 'figure'),
      dep.Output('Summary Table', 'columns'),
      dep.Output('Summary Table', 'data'),
-     dep.Output('Summary Table', 'style_data_conditional'),
      dep.Output('download-link', 'href'),
-     dep.Output('download-link', 'download')],
+     dep.Output('download-link', 'download'),
+     dep.Output('object_threshold', 'children'),
+     dep.Output('object_passed_samples', 'children')],
     [dep.Input('select_a_run', 'value'),
-     dep.Input('lane_select', 'value')])
-def update_graphs(run_alias, lane_alias):
+     dep.Input('lane_select', 'value'),
+     dep.Input('index_threshold', 'value'),
+     dep.Input('pass/fail', 'value'),
+     dep.Input('sample_type', 'value')])
+def update_graphs(run_alias, lane_alias, threshold, PassOrFail, sample_type):
+    index_threshold = int(threshold)
+
     run = df[(df['Run'] == run_alias) & (df['LaneNumber'] == lane_alias)]
-    run = run[~run['library'].isna()].drop_duplicates('library')
+    run = run[~run['Run'].isna()].drop_duplicates('library')
```

Comment:
I don't understand the purpose of this line. `run = df[df['Run'] == run_alias]` above should already take care of this, as long as `run_alias` is not `NaN`.